images/bridge2/webrtc/js/localmedia.js

```diff
@@ -1,15 +1,24 @@
 class LocalMedia {
-  constructor() {
-    this.onstreamchange = (stream) => null;
-    this.ondevicechange = () => null;
-    this.audioDevices = [];
-    this.videoDevices = [];
-    this.outputDevices = [];
-    this.audioSource = undefined;
-    this.videoSource = undefined;
-    this.audioEnabled = true;
-    this.videoEnabled = true;
-    this.stream = undefined;
+  constructor(opts) {
+    let defaults = {
+      onstreamchange: (stream) => null,
+      ondevicechange: () => null,
+      audioDevices: [],
+      videoDevices: [],
+      outputDevices: [],
+      audioSource: undefined,
+      videoSource: undefined,
+      audioEnabled: true,
+      videoEnabled: true,
+      stream: undefined,
+    };
+    for (const [key, defaultValue] of Object.entries(defaults)) {
+      if (opts.hasOwnProperty(key)) {
+        this[key] = opts[key];
+      } else {
+        this[key] = defaultValue;
+      }
+    }
```

Comment:
@progrium any thoughts on a better JS idiom here to initialize the properties?

Reply:
meh, no